dvc/output/base.py

```diff
@@ -156,16 +167,18 @@ def exists(self):
     def changed_checksum(self):
         return (
```

Comment:
Yor `checksum` and `checksum_type` properties, use self.remote checksums, so why not just create `remote.changed_checksum(path_info, checksum_info)` method and use it here? That way all the logic is contained within the remote.

Reply:
You mean `remote.changed(self, path_info, checksum_info):` ?
Because it will raise None Exception in `checksum_to_path` because self.prefix is None.
Tried to use `cache.changed(self, path_info, checksum_info):`
But more Exception cames out, so I think that's why you guys wrote this code before. :)